app/src/main/java/com/naman14/timberx/ui/viewmodels/base/ScopedViewModel.kt

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright (c) 2019 Naman Dwivedi.
+ *
+ * Licensed under the GNU General Public License v3
+ *
+ * This is free software: you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU General Public License for more details.
+ *
+ */
+@file:Suppress("unused", "MemberVisibilityCanBePrivate")
+
+package com.naman14.timberx.ui.viewmodels.base
+
+import androidx.lifecycle.ViewModel
+import kotlinx.coroutines.CoroutineDispatcher
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.CoroutineStart
+import kotlinx.coroutines.Job
+import kotlinx.coroutines.launch
+import org.jetbrains.annotations.TestOnly
+import kotlin.coroutines.CoroutineContext
+
+/** @author Aidan Follestad (@afollestad) */
+abstract class ScopedViewModel(
+    private val mainDispatcher: CoroutineDispatcher
+) : ViewModel() {
+
+    private val job = Job()
```

Comment:
Do you by chance know if this has to be `Job` or `SupervisorJob`? I've seen both.

Reply:
Job is what I use in my production code - and what we've come to use at @square :)

Reply:
ok then i'll trust your words on this :slightly_smiling_face: